oasislmf/pytools/cdftocsv.py

```diff
@@ -0,0 +1,127 @@
+parser.add_argument('--logging-level',
+                    help='logging level (debug:10, info:20, warning:30, error:40, critical:50)', default=30, type=int)
+
+
+def run(run_dir, skip_header, file_in=None, file_out=None):
+    """Run cdftocsv command: convert the cdf output from getmodel into csv format.
+    Presently, the csv file is streamed to stdout.
+
+    Args:
+        run_dir ([str]): Path to the run directory.
+        skip_header ([bool]): If True, does not print the csv header.
+        file_in ([str], optional): If provided, it reads the cdf from that file. Defaults to None.
+        file_out ([str], optional): Placeholder for a future output file writer. Defaults to None.
+
+    Raises:
+        NotImplementedError: If file_out is not None.
+        ValueError: If the stream type is not 1.
+
+    """
+    with ExitStack() as stack:
+        if file_in is None:
+            streams_in = sys.stdin.buffer
+
+        else:
+            streams_in = stack.enter_context(open(file_in, 'rb'))
+
+        if file_out is None:
+            stream_out = sys.stdout
+        else:
+            # implement here a file-writer (e.g., csv writer)
+            raise NotImplementedError("file_out has not been implemented yet.")
+
+        # get damage bins from file
+        static_path = os.path.join(run_dir, 'static')
+        damage_bins = get_mean_damage_bins(static_path=static_path)
+
+        # maximum number of damage bins (individual items can have up to `total_bins` bins)
+        if damage_bins.shape[0] == 0:
+            total_bins = 1000
+        else:
+            total_bins = damage_bins.shape[0]
+
+        # determine stream type
+        stream_type_mv = memoryview(bytearray(4))
+        stream_type = np.ndarray(1, buffer=stream_type_mv, dtype='i4')
+        len_read = streams_in.readinto(stream_type_mv)
+
+        if stream_type[0] != 1:
+            raise ValueError(f"FATAL: Invalid stream type: expect 1, got {stream_type[0]}.")
+
+        if not skip_header:
+            stream_out.write("event_id,areaperil_id,vulnerability_id,bin_index,prob_to,bin_mean")
+
+        # prepare all the data buffers
+        damagecdf_mv = memoryview(bytearray(damagecdfrec.size))
+        damagecdf = np.ndarray(1, buffer=damagecdf_mv, dtype=damagecdfrec.dtype)
+        Nbins_mv = memoryview(bytearray(4))
+        Nbins = np.ndarray(1, buffer=Nbins_mv, dtype='i4')
+        rec_mv = memoryview(bytearray(total_bins * ProbMean.size))
+        rec = np.ndarray(total_bins, buffer=rec_mv, dtype=ProbMean.dtype)
+
+        # start reading the stream
+        # each record from getmodel is expected to contain:
+        # 1 damagecdfrec obj, 1 int (Nbins), a number `Nbins` of ProbMean objects
+        while True:
+            len_read = streams_in.readinto1(damagecdf_mv)
+            len_read = streams_in.readinto1(Nbins_mv)
+            len_read = streams_in.readinto1(rec_mv[:Nbins[0] * ProbMean.size])
```

Comment:
You can't really use readinto1 like this. 
With readinto1 you cannot expect to have values read that is more than 1 byte.
so if for some reason you read less than the size of your mv you will get false results

Reply:
Thanks! I agree: I got mixed up and initially I was using `readinto` which ensure to fill the mv. Then I changed it to `readinto1` following our discussion but `readinto1` was more appropriate here.

I'll now update it with `readinto` so that this is safe, and hopefully can be merged into develop.
I'll then rebase the `feature/gulpy` branch on top of this  so I can work at a version that can be re-used for gulpy.